public/components/trace_analytics/components/common/shared_components/custom_datagrid.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useMemo } from 'react';
+import {
+  EuiDataGrid,
+  EuiButtonEmpty,
+  EuiDataGridColumn,
+  EuiDataGridSorting,
+  EuiOverlayMask,
+  EuiButtonIcon,
+} from '@elastic/eui';
+import { NoMatchMessage } from '../helper_functions';
+
+interface FullScreenWrapperProps {
+  children: React.ReactNode;
+  onClose: () => void;
+  isFullScreen: boolean;
+}
+
+// EUI Data grid full screen button is currently broken, this is a workaround
```

Comment:
nit. Let's add the issue from OUI here as a note for future reference.

Reply:
// OUI Data grid full screen button is currently broken, this is a workaround
// https://github.com/opensearch-project/oui/issues/1166